ipn/ipnlocal/serve.go

```diff
@@ -254,10 +268,10 @@ func (b *LocalBackend) setServeConfigLocked(config *ipn.ServeConfig) error {
 
 // ServeConfig provides a view of the current serve mappings.
 // If serving is not configured, the returned view is not Valid.
-func (b *LocalBackend) ServeConfig() ipn.ServeConfigView {
+func (b *LocalBackend) ServeConfig() (ipn.ServeConfigView, string) {
 	b.mu.Lock()
 	defer b.mu.Unlock()
-	return b.serveConfig
+	return b.serveConfig, b.serveConfigETag
```

Comment:
how about we just define that the etag is the sha256 of the JSON-ified result? So we don't need to store it and return it, etc?

Reply:
@bradfitz sounds good, do you mean to have clients calculate the etag on their side, or that the LocalBackend still sets the etag header but just on the fly by the handler and not as a persisted field? 

I'm going with the latter, but double checking.

Reply:
Yes, the latter. The client should still remember it from a get and send it along explicitly in a conditional put later.